alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDeleteMonitorAction.kt

```diff
@@ -104,9 +106,10 @@ class TransportDeleteMonitorAction @Inject constructor(
             try {
                 val monitor = getMonitor()
 
-                val canDelete = user == null ||
-                    !doFilterForUser(user) ||
-                    checkUserPermissionsWithResource(user, monitor.user, actionListener, "monitor", monitorId)
+                val canDelete = monitorIsNotInWorkflows(monitor.id) && (
```

Comment:
If the monitor cannot be deleted if it is part of a workflow, we need to mention it is part of a workflow and that is why it cannot be deleted.